src/main/scala/beam/agentsim/agents/ridehail/RideHailAgent.scala

```diff
@@ -228,6 +234,64 @@ class RideHailAgent(
           .copy(currentVehicle = Vector(vehicle.id), remainingShifts = shifts.get)
       }
   }
+
+  when(PendingOfflineForCharging) {
```

Comment:
This interim step was added only because agents were getting stuck otherwise. It matches the flow of what happens without adding my logic. My biggest concern is how tick and trigger id are handled since the RHA is not scheduling this, but instead handling it itself. In a sf-light .5k run I encountered 40 

> Current tick is empty when moving vehicle to OfflineForCharging. This likely will get the agent stuck

However none got stuck like previously, so maybe it is fine. But if there is a better, more seamless/direct way to do this than using onTransition and this pending state then I'm all for it.